tests/langtest_c.rs

```diff
@@ -11,6 +14,30 @@ use tests::{mk_compiler, EXTRA_LINK};
 use ykbuild::{completion_wrapper::CompletionWrapper, ykllvm_bin};
 
 const COMMENT: &str = "//";
+const COMMENT_PREFIX: &str = "##";
+
+/// Parse any "extra environment" to pass to yk-config out of the test file.
+fn parse_yk_config_env(p: &Path) -> Result<HashMap<String, String>, Box<dyn Error>> {
+    let f = File::open(p)?;
+    let rdr = BufReader::new(&f);
+    let mut extra = HashMap::new();
+    let env_prefix = format!("{} {} yk-config-env:", COMMENT, COMMENT_PREFIX);
+    for line in rdr.lines() {
+        let line = line?;
+        if !line.starts_with(COMMENT) {
+            break; // won't find any lower down the file.
+        }
+        if line.starts_with(&env_prefix) {
+            let sfx = &line[(&env_prefix).len()..];
+            let mut elems = sfx.split("=");
+            extra.insert(
+                elems.next().ok_or("bad env spec")?.trim().into(),
```

Comment:
`bad env spec` is a bit obtuse: I'd prefer either an `unwrap` (so someone will look at this line of code and then suss out the fix) or a descriptive error message.